framework/include/cppmicroservices/BundleContext.h

```diff
@@ -668,11 +826,11 @@ class US_Framework_EXPORT BundleContext
    * @see RemoveServiceListener()
    */
   template<class R>
-  void AddServiceListener(R* receiver, void(R::*callback)(const ServiceEvent&),
-                          const std::string& filter = std::string())
+  ListenerToken AddServiceListener(R* receiver, void(R::*callback)(const ServiceEvent&),
```

Comment:
Agreed, this makes it a single interface to add listeners for the clients. 

Note that as per the design document, currently, this is the only instance where if an attempt is made to add a listener (with or without a different filter) of this type to the system, it'll replace the existing listener in the system. 

The motivation was that the existing client code functionality should work as intended, unless it's an obvious bug which is getting fixed in this PR. Not doing this would mean a lot of tests such as ServiceHooksTest would have to change. Please refer to the design doc for more information.